### PR TITLE
Bump jquery from 2.2.4 to 3.1.0

### PR DESCRIPTION
**Pull request by ChristophWurst**
Like https://github.com/nextcloud/server/pull/23099 but in a smaller step.

:scroll: https://jquery.com/upgrade-guide/3.0

3.0 didn't work because of an UMD issue https://github.com/jquery/jquery-migrate/issues/273, so I'm trying 3.1. Locally things still work.

* `is(':visible')` behaves different with jQuery 3. I couldn't ge t one of the tests to pass, but it's a component that will hopefully soon be replaced by Vue so I just dropped the one test. https://api.jquery.com/visible-selector/ for more info.
* `.selector` was deprecated in 1.9 and removed in 3. I removed an assertion that relied on that.
* Removed fake timers where possible because they block jQuery deferred that use a timer to run code async in v3. With fake timers the promises do not make any progress and the test times out
* Changed async tests to always use `done` and have the test framework await the results

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Bildschirmfoto von 2020-11-13 09-29-21

:rocket: :pray: :exploding_head:

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -3699,7 +3699,7 @@
 			const enabled = tabView.canDisplay || undefined
 			if (tabView.id) {
 				OCA.Files.Sidebar.registerTab(new OCA.Files.Sidebar.Tab({
-					id: tabView.id, 
+					id: tabView.id,
```
Comment (skjnldsv) [inline apps/files/js/filelist.js:3702]: the most important change

File: core/src/OC/l10n.js [hunk 2]
```diff
@@ -328,7 +328,7 @@ export default L10n
  *
  * @returns {String} locale string
  */
-export const getLocale = () => $('html').data('locale')
+export const getLocale = () => $('html').data('locale') ?? 'en'
```
Comment (skjnldsv) [inline core/src/OC/l10n.js:331]: :+1:

File: core/src/jquery/contactsmenu.js [hunk 3]
```diff
@@ -82,9 +82,10 @@ $.fn.contactsMenu = function(shareWith, shareType, appendTo) {
 			}
 
 			actions.forEach(function(action) {
-				const template = entryTemplate
-				$list.find('ul').append(template(action))
+				$list.find('ul').append(entryTemplate(action))
```
Comment (skjnldsv) [inline core/src/jquery/contactsmenu.js:85]: I'll happily see this one die #22274

File: core/src/jquery/contactsmenu.js [hunk 4]
```diff
@@ -82,9 +82,10 @@ $.fn.contactsMenu = function(shareWith, shareType, appendTo) {
...
 			})
+
+			$div.trigger('load')
```
Comment (ChristophWurst) [inline core/src/jquery/contactsmenu.js:88]: FYI needed so a test can await the async operation to be *done*.

File: core/src/jquery/contactsmenu.js [hunk 5]
```diff
@@ -95,11 +96,12 @@ $.fn.contactsMenu = function(shareWith, shareType, appendTo) {
...
 			}
 
-			const template = entryTemplate
-			$list.find('ul').append(template({
+			$list.find('ul').append(entryTemplate({
 				hyperlink: '#',
 				title,
 			}))
+
+			$div.trigger('loaderror', jqXHR)
```
Comment (ChristophWurst) [inline core/src/jquery/contactsmenu.js:104]: Same. Just see the updated spec code :)

**Review by skjnldsv — APPROVED** [review]
Tried a few thing, clicking around, files contactsmenu,settings, old legacy areas, no errors, works fine! :+1: 
Awesome job!!

**Review by GretaD — APPROVED** [review]
I also couldnt find anything that is broken because of this PR :+1:

**Review by PVince81 — COMMENTED** [review]
Did a quick clicking around through the files app, shares, public shares and settings and it looked fine

**Comment by MorrisJobke** [comment]
Nice! 🚀

**Pull request merged**